cartography/intel/gcp/storage.py

```diff
@@ -0,0 +1,223 @@
+                 "Full details: %s"), bucket, e, )
+            return None
+        else:
+            raise
+
+
+def transform_gcp_buckets(bucket_res): 
+    '''
+    Transform the GCP Storage Bucket response object for Neo4j ingestion
+    
+    :type bucket_res: A storage resource object (https://cloud.google.com/storage/docs/json_api/v1/buckets) 
+    :param bucket_res: The return data
+    
+    :rtype: list
+    :return: List of buckets ready for ingestion to Neo4j
+    '''
+    
+    bucket_list = []
+    for b in bucket_res.get('items', []):
+        bucket = {}
+        bucket['etag'] = b.get('etag', '')  
+        bucket['iam_config_bucket_policy_only'] = b.get('iamConfiguration'
+                                                        , {}).get('bucketPolicyOnly', {}).get('enabled', None)
+        bucket['iam_config_uniform_bucket_level_access'] = b.get('iamConfiguration'
+                                                                 , {}).get('uniformBucketLevelAccess', {}).get('enabled', None)
+        bucket['id'] = b.get('id', '') 
+        bucket['labels'] = [(key, val) for (key, val) in b.get('labels', {}).items()] 
+        bucket['owner_entity'] = b.get('owner', {}).get('entity', '') 
+        bucket['owner_entity_id'] = b.get('owner', {}).get('entityId', '') 
+        bucket['kind'] = b.get('kind', '') 
+        bucket['location'] = b.get('location', '') 
+        bucket['location_type'] = b.get('locationType', '')
+        bucket['meta_generation'] = g.get('metageneration', None) 
+        bucket['project_number'] = b.get('projectNumber', None) 
+        bucket['self_link'] = b.get('selfLink', '') 
+        bucket['storage_class'] = b.get('storageClass', '')
+        bucket['time_created'] = b.get('timeCreated', '') 
+        bucket['updated'] = b.get('updated', '') 
+        bucket['versioning_enabled'] = b.get('versioning', {}).get('enabled', None) 
+        bucket['default_event_based_hold']  = b.get('defaultEventBasedHold', None) 
+        bucket['retention_period'] = b.get('retentionPolicy', {}).get('retentionPeriod', None) 
+        bucket['default_kms_key_name'] = b.get('encryption', {}).get('defaultKmsKeyName', '') 
+        bucket['log_bucket'] = b.get('logging', {}).get('logBucket', '') 
+        bucket['requester_pays'] = b.get('billing', {}).get('requesterPays', None)  
+        bucket_list.append(bucket)
+    return bucket_list
+
+
+def load_gcp_buckets(neo4j_session, buckets, gcp_update_tag): 
+    '''
+    Ingest GCP Storage Buckets to Neo4j
+    
+    :type neo4j_session: Neo4j session object 
+    :param neo4j session: The Neo4j session object
+    
+    :type buckets: list
+    :param buckets: List of GCP Buckets to injest 
+    
+    :type gcp_update_tag: timestamp
+    :param gcp_update_tag: The timestamp value to set our new Neo4j nodes with
+    
+    :rtype: NoneType
+    :return: Nothing 
+    '''
+    
+    query = """
```

Comment:
Might be tricky to handle `labels` here :-).  I _think_ we can do a list type on the node: https://neo4j.com/docs/cypher-manual/current/functions/list/

Reply:
[Seems like](https://stackoverflow.com/a/58229056/10981911) you can pass them as any other parameter. Also, should probably start migrating our neo4j parameter syntax from `{param}` to `$param` since the former is [depreciated](https://neo4j.com/docs/cypher-manual/current/syntax/parameters/), but that should probably be a topic for another PR.

Reply:
> You can pass them as any other parameter

Sounds good.

> Also, we should probably start migrating our neo4j parameter syntax from {param} to $param since the former is depreciated, but that should probably be a topic for another PR.

Boo :|  Thanks for pointing that out.

> Also, there seems to be multiple instances of nested SET clauses, but I thought that wasn't allowed?

`ON CREATE SET` and `SET` are two separate clauses.  We call `ON CREATE SET` when making the `firstseen` attribute, and we call `SET` when updating the other fields of a node.